### ISSUE TITLE
Support for lazy property synchronization

### ISSUE THREAD

**Issue by EelcoHoogendoorn** [zoofIO/flexx#289]
A potentially generally useful feature occurred to me while profiling our flexx app; when editing large properties (large arrays) on the client, the fluididity of the interaction can become limited by the serialization and communication to the server.

If we had something like @event.lazyprop, which would only sychronize on-demand, as the property would be accessed, this kind of overhead may be avoided in many scenarios. In our case, we are only interested in reading out such arrays server-side once the client is done applying its numerous edits.

**Comment by almarklein** [comment]
Synchronizing on-demand would be a bit tricky, because you don't want the ui thread to wait while the syncing is performed. So it'd have to be a scenario where the client says "I am now interested in the data", and then the syncing would be performed async, generating an event when the client has the latest version.

You can do this by using events instead of properties, though its a bit more work (and this problem is what we're trying to solve now with DataSources).

I think this problem is specific to syncing "data". I'd love to have builtin support to send data to the client, but I am not sure if the API for it should look like a property.

**Comment by almarklein** [comment]
I think this is really a duplicate of #143, do you agree?

**Comment by almarklein** [comment]
Closing this, as I think `send_data()` (and datasources in our project) cover most use-cases, and also because its hard to implement this due to the async nature of Flexx. Can be reopened if the need persists, but I have a feeling this is a problem that should be fixed in another way.
